### DISCUSSION TITLE
Control Panel logs should record account deletions

### DISCUSSION THREAD

**Discussion by nep** [ExpressionEngine/ExpressionEngine#3392] [Feature Requests]
**Is your feature request related to a problem? Please describe.**

This is a feature request that I consider a borderline bug because of the security implications.

When you create a member account, it is recorded in the Control Panel Logs.

But when you delete an account, that action is not recorded in the Control Panel Logs.  So you can't see when an account was deleted or by who.

Furthermore, when you delete an account, all actions done by that account are removed from the control panel logs. Which means you can't tell if an account did something before they were deleted which, in a recent case, made it very difficult to tell if a security breach had occurred and in what way.

**Describe the solution you'd like**

Deleting an account should be tracked in the Control Panel log just like account creation.

Actions by that deleted account prior to deletion should be anonymized but preserved. The account ID itself could be sufficient, and is not tied to PII once the member account is gone -- or even generate a pseudo account ID, like a random negative number. 

But not being able to tell even that a deletion occurred, that's not cool.

(Also consider, when an account is deleted, the CP log still lists when that account was originally created and with what user name. Just another thread to consider.)

**Comment by nep** [comment]
Having just encountered this painfully, and remembering that I've had this same situation multiple before, can I extend this to ask that the CP log also record all Entry deletions and by who?

**Comment by intoeetive** [comment]
@nep this is on our radars - thank you!

**Comment by intoeetive** [comment]
https://github.com/ExpressionEngine/ExpressionEngine/pull/3805
